solutions/src/main/java/ru/sberbank/school/task02/Solution.java

```diff
@@ -1,29 +1,77 @@
 package ru.sberbank.school.task02;
 
-import java.math.BigDecimal;
-
-import ru.sberbank.school.task02.exception.FxConversionException;
-import ru.sberbank.school.task02.util.ClientOperation;
-import ru.sberbank.school.task02.util.ExternalQuotesServiceDemo;
-import ru.sberbank.school.task02.util.Symbol;
+import ru.sberbank.school.task02.util.*;
+import ru.sberbank.school.task02.util.FxRequest;
+import ru.sberbank.school.task02.util.FxResponse;
 
 public class Solution {
     public static void main(String[] args) {
 
         CalculatorFactory calculatorFactory = new CalculatorFactory();
-        // получаем наш валютный калькулятор
         FxConversionService calculator =
                 calculatorFactory.getFxConversionService(new ExternalQuotesServiceDemo());
+        Client client = new Client(calculator);
+
+        FxRequest request = parseArgs(args);
+        FxResponse response = client.fetchResult(request);
+        System.out.println(response.toString());
+
+    }
+
+    private static FxRequest parseArgs(String[] args) {
+
+        if (args.length == 0) {
+            showUsage();
+            System.exit(0);
+        }
+
+        String symbol = null;
+        String operation = null;
+        String amount = null;
+
+        int i = 0;
 
         try {
-            BigDecimal price = calculator.convert(ClientOperation.BUY,
-                    Symbol.USD_RUB, BigDecimal.valueOf(1000));
-            System.out.println(price);
-        } catch (FxConversionException ex) {
-            System.out.println(ex.getMessage());
-        } catch (NullPointerException ex) {
-            System.out.println(ex.getMessage());
+            while (i < args.length) {
+                if (args[i].equalsIgnoreCase("--symbol")) {
+                    symbol = args[i + 1];
+                    i += 2;
+                } else if (args[i].equalsIgnoreCase("--operation")) {
+                    operation = args[i + 1];
+                    i += 2;
+                } else if (args[i].equalsIgnoreCase("--amount")) {
+                    amount = args[i + 1];
+                    i += 2;
+                } else {
+                    System.out.println("Unknows parameter: " + args[i]);
+                    showUsage();
+                    System.exit(1);
+                }
+            }
+        } catch (IndexOutOfBoundsException ex) {
+            System.out.println("No value for parameter: " +args[i]);
```

Comment:
WhitespaceAround: &#39;+&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a229852653>*